### ISSUE TITLE
1.6.7rc4 - Problem with tree order

### ISSUE THREAD

**Issue by gnerred** [camptocamp/c2cgeoportal#1803]
If we add or remove a layer from a layergroup, all order values are reseted... The values entered manually are not kept. I saw another issue talking about that, but not sure if it is the same problem, because it had to be fixed in 1.6.7.

**Comment by gnerred** [comment]
Any comment about this issue ?

**Comment by gnerred** [comment]
@sbrunner , could you please give me an answer about this issue ?

**Comment by sbrunner** [comment]
It's not an issue it's how the order actually work, in prevision of the new admin tool who should completely hide this value.

**Comment by gnerred** [comment]
If it's not an issue, it's a real problem... Here is a use case : 

1°/ Create a layergroup named "ttt"
2°/ Integrate layers "aaa","bbb" and "ccc" in this layergroup
3°/ Change the order of each layer in "Ordre des liens dans l'arbre", because I don't want the order proposed by default
4°/ One month later, I have to add a new layer named "ddd" in this layergroup "ttt"
5°/ First problem : in some cases I have to reselect layers "aaa","bbb" and "ccc". If I select only "ddd", the three other layers disappear from my layergroup
6°/ If I reselect "aaa","bbb" and "ccc", dans then select the new layer "ddd", all the orders defined manuelly are "fucked", and I have to redifine all manually. 

So if this behaviour is not a bug, it makes the admin interface not usable when dealing with layergroups ! We loose an incredible time each time we have to modify the content of a layergroup. 

Here are my pain friends : @rbovard , @kalbermattenm , @wunderlichr 

So it will be great if something could be done, otherwise all the integrators will have the same remark...

**Comment by rbovard** [comment]
+1 with @gnerred 

It's also a real problem for people that using order value to maintain all layers / groups / themes together, see #1623 

> 5°/ First problem : in some cases I have to reselect layers "aaa","bbb" and "ccc". If I select only "ddd", the three other layers disappear from my layergroup

It's a cache issue (also present in 1.5), refresh the page with F5 and the layers will be checked.

**Comment by kalbermattenm** [comment]
We have a similar issue...

It is quite messy for us, because if the order of elements, might it be layers or groups, a "recomputed" each time we make a change, then we have a huge issue. At SITN, there are so many interdependent elements that this cannot be automated. A layer might be part of several groups which are part of several other groups in different themes. This has to be done manually (I cannot imagine a way to automate that).

**Comment by gnerred** [comment]
thanks @rbovard for tip

**Comment by sbrunner** [comment]
I think that °6 is also related on °5. I never be able to reproduce it...

**Comment by gnerred** [comment]
Strange that you cannot reproduce, because Neuchâtel, Nyon, Jura and Cartoriviera have the same problem...

**Comment by kalbermattenm** [comment]
I just tried it out... and it is an issue... I did that using pserve.

* First I created a group `toto`
* I added three existing layers to this group
* I change the layer order for this group in the "Ordre des liens dans l'arbre", because I was not happy with the proposed default order.
* I went back to the group admin interface, double-clicked on `toto`
* I added a fourth layer and saved

The result is that all my previous layers which were bound to `toto` have disappeared... There are no more bound to `toto`...

And this is quite annoying...

**Comment by gnerred** [comment]
This is the problem I mentioned on point 5. Can you reproduce my bug on point 6 ? Reselecting manually previous layers, so that the result is that you add a new layer in the group.

**Comment by kalbermattenm** [comment]
Yes, I have the same problem...

The problem is that when you open an existing group, then the interface does not show out which layers are already part of the group. If you add a layer without re-selecting the already linked layers, then you loose all the previous layers of the group.

By re-selecting them in order not to loose them, the GMF admin methods and function think that you are recreating a group (or re-adding all its children layers/groups) and it recreates the order stuff from scratch.

When I added four layers to a group, in the "Ordre des liens de l'arbre" UI they had ids 626, 627, 628, 629.

Then I did as you suggested in your point 6 and added a new layer to this group (while re-selecting also the fourth first layers).  I then added a fifth layer and the ids are now 630, 631, 632, 633, 634, thus new entries...

This is clearly an issue...

@sbrunner, I would be happy to make you a demo (hangout) if needed...
